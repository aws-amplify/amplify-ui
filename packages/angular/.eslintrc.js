module.exports = {
  root: true,
  overrides: [
    {
      files: ['**/*.ts'],
      parserOptions: {
        project: ['tsconfig.json'],
        tsconfigRootDir: __dirname,
        createDefaultProgram: true,
      },
      extends: [
        'plugin:@typescript-eslint/recommended',
        'plugin:@angular-eslint/recommended',
        // This is required if you use inline templates in Components
        'plugin:@angular-eslint/template/process-inline-templates',
        '@aws-amplify/amplify-ui',
      ],
      rules: {
        /*
         * `prefer-nullish-coalescing` requires `strictNullChecks` to be on.
         * Turn this off until we turn `strict` on in the `tsconfig.json`.
         */
        '@typescript-eslint/prefer-nullish-coalescing': 'off',

        // we use `amplify-` prefix for directives.
        '@angular-eslint/directive-selector': [
          'error',
          {
            type: 'attribute',
            prefix: 'amplify',
            style: 'camelCase',
          },
        ],
      },
    },
    {
      files: ['**/*.html'],
      extends: ['plugin:@angular-eslint/template/recommended'],
    },
    {
      extends: ['@aws-amplify/amplify-ui/jest'],
      files: ['**/__mocks__/**', '**/__tests__/**'],
    },
  ],
};
