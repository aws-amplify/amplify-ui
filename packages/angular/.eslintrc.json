{
  "root": true,
  "ignorePatterns": ["projects/**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "parserOptions": {
        "project": ["tsconfig.json"],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "amplify",
            "style": "camelCase"
          }
        ]
      }
    },
    {
      "files": ["*.html"],
      "extends": ["plugin:@angular-eslint/template/recommended"],
      "rules": {}
    }
  ]
}
