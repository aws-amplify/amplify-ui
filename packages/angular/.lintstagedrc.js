const config = require('../../.lintstagedrc.js');

module.exports = {
  ...config,
  '*.ts': 'eslint',
  '**/*.ts': 'tsc-files --noEmit',
};
