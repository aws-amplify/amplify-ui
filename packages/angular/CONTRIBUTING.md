# Contributing

## Development Structure

`@aws-amplify/ui-angular` utilizes a monorepo structure to manage devDependencies and the actual packages separately.

This directory (`packages/angular`) contains base dependencies for developing angular packages. The actual code for `@aws-amplify/ui-angular` is located in [`projects/ui-angular`](./projects/ui-angular/) directory.

```bash
packages/angular/
├── projects/ui-angular/ # where the actual `@aws-amplify/ui-angular` library lives
│   ├── src/ # contains src files for UI library code.
│   └── package.json # contains dependencies that library needs (e.g. xstate)
│
├── dist/ui-angular/ # this is auto-generated on build. Deployments will only happen from this folder.
│   └── esm2015/
│   └── fesm2015/
│   └── lib/
│   └── package.json # same as the above, but defines extra fields like
│                    # main, module, esm2015, typings, sideEffects.
│
└── package.json # contains base dependencies for developing angular packages.
                 # This is only for infrastructure purpose! Shouldn't be
                 # published or be referenced by other dependencies.

```

## Getting Started

1. Navigate to the _root_ of your local clone of [aws-amplify/amplify-ui](https://github.com/aws-amplify/amplify-ui)
1. Run `yarn setup`
1. Run `yarn angular dev`

This will start building `@aws-amplify/ui-angular` in watch mode. To test your changes, you can utilize [`examples/angular`](../../examples/angular) to run examples on Angular. Please see examples [README](../../examples/README.md) and e2e [README](../e2e/README.md#contributing) to get started.

### Depenencies

`@aws-amplify/ui-react` depends on [`@aws-amplify/ui`](../ui) for theming, state management, and translation logic. If you're looking for change in these, please refer to `@aws-amplify/ui` [README](../ui/README.md).

## Testing for Production

After you tested your change, you can run `yarn angular build` and `yarn angular-example build` from monorepo to run build for production.

## Troubleshooting

If you're cross developing `angular` and `angular-example`, make sure you're running the same type of commands -- either run `yarn angular dev` and `yarn angular-example dev` together, or run `yarn angular build` and `yarn angular-example build` together. Not doing so will result in IVY compiler conflict.
