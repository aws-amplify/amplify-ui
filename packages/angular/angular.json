{
  "$schema": "../../node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "ui-angular": {
      "projectType": "library",
      "root": "projects/ui-angular",
      "sourceRoot": "projects/ui-angular/src",
      "prefix": "amplify",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "tsConfig": "projects/ui-angular/tsconfig.lib.json",
            "project": "projects/ui-angular/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/ui-angular/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/ui-angular/**/*.ts",
              "projects/ui-angular/**/*.html"
            ]
          }
        }
      }
    }
  },
  "cli": {
    "schematicCollections": [
      "@angular-eslint/schematics"
    ],
    "analytics": false
  }
}
