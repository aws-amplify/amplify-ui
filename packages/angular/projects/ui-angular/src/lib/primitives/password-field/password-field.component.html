<div
  [class]="
    classnames(
      ComponentClassName.Flex,
      ComponentClassName.Field,
      ComponentClassName.TextField,
      ComponentClassName.PasswordField
    )
  "
>
  <label
    class="amplify-label"
    [class.amplify-visually-hidden]="labelHidden"
    [for]="fieldId"
  >
    {{ label }}
  </label>
  <div
    [class]="classnames(ComponentClassName.Flex, ComponentClassName.FieldGroup)"
  >
    <div [class]="ComponentClassName.FieldGroupFieldWrapper">
      <input
        [class]="
          classnames(
            ComponentClassName.Input,
            ComponentClassName.FieldGroupControl,
            ComponentClassName.FieldShowPassword
          )
        "
        [id]="fieldId"
        [type]="type"
        [name]="name"
        [placeholder]="placeholder"
        [required]="required"
        [value]="initialValue"
        [autocomplete]="autocomplete"
        autocapitalize="off"
        [attr.disabled]="disabled ? '' : null"
        [attr.aria-invalid]="hasError ? 'true' : 'false'"
        [attr.aria-describedby]="describedBy"
        (blur)="setBlur.emit($event)"
      />
    </div>
    <div [class]="ComponentClassName.FieldGroupOuterEnd">
      <button
        amplify-button
        role="switch"
        (click)="togglePasswordText()"
        [attr.aria-checked]="type !== 'password'"
        [attr.aria-label]="showPasswordButtonlabel"
        [class]="
          classnames(
            ComponentClassName.Input,
            ComponentClassName.FieldGroupControl
          )
        "
      >
        <svg
          *ngIf="!showPassword"
          xmlns="http://www.w3.org/2000/svg"
          fill="currentColor"
          viewBox="0 0 24 24"
          class="amplify-icon"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          ></path>
        </svg>
        <svg
          *ngIf="showPassword"
          xmlns="http://www.w3.org/2000/svg"
          fill="currentColor"
          viewBox="0 0 24 24"
          class="amplify-icon"
        >
          <path
            d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
            fill="none"
          ></path>
          <path
            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
          ></path>
        </svg>
      </button>
    </div>
  </div>
</div>
