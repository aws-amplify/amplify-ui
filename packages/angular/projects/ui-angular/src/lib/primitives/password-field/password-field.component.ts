import { Component, EventEmitter, Input, Output } from '@angular/core';
import { translate, ComponentClassName } from '@aws-amplify/ui';
import { nanoid } from 'nanoid';
import classnames from 'classnames';

@Component({
  selector: 'amplify-password-field',
  templateUrl: './password-field.component.html',
})
export class PasswordFieldComponent {
  @Input() autocomplete = 'new-password';
  @Input() disabled = false;
  @Input() fieldId = `amplify-field-${nanoid(12)}`;
  @Input() initialValue = '';
  @Input() label = '';
  @Input() name: string;
  @Input() placeholder = '';
  @Input() required = true;
  @Input() labelHidden = false;
  @Input() hasError: boolean;
  @Input() describedBy: string;
  @Output() setBlur = new EventEmitter<Event>();

  public type: 'text' | 'password' = 'password';

  public showPassword = false;
  public showPasswordButtonlabel = translate('Show password');

  // re-export utilities so that template html can use them
  public classnames = classnames;
  public ComponentClassName = ComponentClassName;

  togglePasswordText(): void {
    this.showPassword = !this.showPassword;
    this.showPasswordButtonlabel = this.showPassword
      ? translate('Hide password')
      : translate('Show password');
    this.type = this.showPassword ? 'text' : 'password';
  }
}
