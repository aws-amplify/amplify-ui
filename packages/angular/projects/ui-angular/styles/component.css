[data-amplify-authenticator] {
  font-size: 1rem;
  line-height: 1.5rem;
  --tw-text-opacity: 1;
  color: rgba(55, 65, 81, var(--tw-text-opacity));
  --tw-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000),
    var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
  font-family: 'Work Sans', Arial, Helvetica, sans-serif;
}

[data-amplify-wrapper] [data-amplify-heading] {
  margin-bottom: 0px;
  font-size: 1.25rem;
  line-height: 1.75rem;
}

[data-amplify-authenticator] [data-amplify-form] {
  display: grid;
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  padding: 2rem;
}

[data-amplify-wrapper] [data-amplify-label] > :not([hidden]) ~ :not([hidden]) {
  --tw-space-y-reverse: 0;
  margin-top: calc(0.5rem * calc(1 - var(--tw-space-y-reverse)));
  margin-bottom: calc(0.5rem * var(--tw-space-y-reverse));
}

[data-amplify-confirmationcode] {
  display: grid;
}

[data-amplify-wrapper] [data-amplify-password] [data-amplify-box] {
  display: flex;
  flex-direction: row;
}

[data-amplify-wrapper]
  [data-amplify-password]
  [data-amplify-box]
  > :not([hidden])
  ~ :not([hidden]) {
  --tw-space-x-reverse: 0;
  margin-right: calc(0.25rem * var(--tw-space-x-reverse));
  margin-left: calc(0.25rem * calc(1 - var(--tw-space-x-reverse)));
}

[data-amplify-wrapper]
  [data-amplify-password]
  [data-amplify-box]
  [data-amplify-text] {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
}

[data-amplify-wrapper]
  [data-amplify-password]
  [data-amplify-box]
  [data-amplify-button] {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(59, 130, 246, var(--tw-text-opacity));
}

[data-amplify-wrapper] [data-amplify-footer] [data-amplify-text] {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
}

[data-amplify-wrapper]
  [data-amplify-footer]
  [data-amplify-button]:first-of-type {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(59, 130, 246, var(--tw-text-opacity));
}

[data-amplify-wrapper]
  [data-amplify-footer]
  [data-amplify-button]:last-of-type {
  border-radius: 0.25rem;
  --tw-bg-opacity: 1;
  background-color: rgba(55, 65, 81, var(--tw-bg-opacity));
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
}

[data-amplify-authenticator] :disabled {
  opacity: 0.5;
  cursor: initial;
}

[data-ui-error] {
  color: orangered;
}

[data-amplify-wrapper] [data-amplify-federated] {
  display: flex;
  width: 100%;
  flex-direction: column;
  gap: 1rem;
}

[data-amplify-wrapper] [data-amplify-federated] [data-amplify-button] {
  border-radius: 0.5rem;
  background-color: transparent;
  padding: 0.5rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(59, 130, 246, var(--tw-text-opacity));
}

[data-amplify-button] {
  cursor: pointer;
}

[data-amplify-strike] {
  width: 100%;
  text-align: center;
  border-bottom: 1px solid #c4c4c4;
  line-height: 0.1em;
  margin: 32px 0;
  color: #828282;
}

[data-amplify-strike] [data-amplify-spacer] {
  background: white;
  padding: 0 25px;
  font-size: 14px;
  font-weight: 500;
}

[data-amplify-wrapper] [data-amplify-verify-label] {
  display: flex;
}

[data-amplify-wrapper] [data-amplify-verify-input] {
  width: initial;
}

.federated-sign-in-button {
  border-color: black;
  border-radius: 0;
  color: black;
}

.federated-sign-in-button-row {
  position: relative;
}

.federated-sign-in-icon {
  position: absolute;
  left: 0;
}
