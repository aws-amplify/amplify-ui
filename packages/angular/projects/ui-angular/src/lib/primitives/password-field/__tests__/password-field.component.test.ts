import { ComponentClassName } from '@aws-amplify/ui';
import { screen, render, fireEvent } from '@testing-library/angular';

import { PasswordFieldComponent } from '../password-field.component';

describe('amplify-password-field', () => {
  it('renders as expected', async () => {
    const { container } = await render(PasswordFieldComponent, {
      componentProperties: {
        name: 'password',
        label: 'Password',
        fieldId: 'mockId',
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('renders as expected when show password button is clicked', async () => {
    const { container } = await render(PasswordFieldComponent, {
      componentProperties: {
        name: 'password',
        label: 'Password',
        fieldId: 'mockId',
      },
    });
    const showPasswordButton = screen.getByRole('switch');
    fireEvent.click(showPasswordButton);

    expect(screen.getByLabelText('Hide password')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render with default classnames', async () => {
    await render(
      `
        <amplify-password-field label="Password" data-testid="test"></amplify-password-field>
      `,
      {
        declarations: [PasswordFieldComponent],
      }
    );

    // Angular components always have an empty wrapper. Go to its first
    // child to reach the actual container.
    const passwordField = screen.getByTestId('test').firstChild;

    expect(passwordField).toHaveClass(ComponentClassName.Flex);
    expect(passwordField).toHaveClass(ComponentClassName.Field);
    expect(passwordField).toHaveClass(ComponentClassName.TextField);
    expect(passwordField).toHaveClass(ComponentClassName.PasswordField);
  });

  it('should render label with expected classname', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: { name: 'password', label: 'Password' },
    });
    const label = screen.getByText('Password');
    expect(label).toHaveClass(ComponentClassName.Label);
  });

  it('should hide label when labelHidden is true', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: {
        name: 'password',
        label: 'Password',
        labelHidden: true,
      },
    });
    const label = screen.getByText('Password');
    expect(label).toHaveClass('amplify-visually-hidden');
  });

  it('should render input with expected classnames', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: {
        label: 'Password',
        fieldId: 'mockId',
        labelHidden: true,
      },
    });

    const input = screen.getByLabelText('Password');
    expect(input.tagName).toBe('INPUT');
    expect(input).toHaveClass(ComponentClassName.Input);
    expect(input.id).toBe('mockId');
  });

  it('should render input with autocomplete enabled by default', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: { name: 'password', label: 'Password' },
    });
    const input = screen.getByLabelText('Password');
    expect(input).toHaveAttribute('autocomplete', 'new-password');
  });

  it('should have aria-invalid when error is present', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: {
        name: 'password',
        label: 'Password',
        hasError: true,
      },
    });
    const input = screen.getByLabelText('Password');
    expect(input).toHaveAttribute('aria-invalid', 'true');
  });

  it('should have aria-describedby when describedBy is present', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: {
        name: 'password',
        label: 'Password',
        describedBy: 'error-id',
      },
    });
    const input = screen.getByLabelText('Password');
    expect(input).toHaveAttribute('aria-describedby', 'error-id');
  });

  it('should require input when required is true', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: {
        name: 'password',
        label: 'Password',
        required: true,
      },
    });
    const input = screen.getByLabelText('Password');
    expect(input).toHaveAttribute('required');
  });

  it('should disable input when disabled is true', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: {
        name: 'password',
        label: 'Password',
        disabled: true,
      },
    });
    const input = screen.getByLabelText('Password');
    expect(input).toBeDisabled();
  });

  it('toggles input field type when show password button is clicked', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: { name: 'password', label: 'Password' },
    });
    const input = screen.getByLabelText('Password');
    const showPasswordButton = screen.getByRole('switch');

    expect(input).toHaveAttribute('type', 'password');

    fireEvent.click(showPasswordButton);
    expect(input).toHaveAttribute('type', 'text');

    fireEvent.click(showPasswordButton);
    expect(input).toHaveAttribute('type', 'password');
  });

  it('reflects default value', async () => {
    await render(PasswordFieldComponent, {
      componentProperties: {
        name: 'password',
        label: 'Password',
        labelHidden: true,
        initialValue: 'initialvalue',
      },
    });
    const input: HTMLInputElement = screen.getByLabelText('Password');
    expect(input.value).toBe('initialvalue');
  });
});
