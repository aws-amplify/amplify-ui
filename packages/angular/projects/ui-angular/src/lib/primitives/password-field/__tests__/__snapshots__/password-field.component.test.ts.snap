// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-password-field renders as expected 1`] = `
<div
  id="root0"
>
  <div
    class="amplify-field amplify-flex amplify-passwordfield amplify-textfield"
  >
    <label
      class="amplify-label"
      for="mockId"
    >
       Password 
    </label>
    <div
      class="amplify-field-group amplify-flex"
    >
      <div
        class="amplify-field-group__field-wrapper"
      >
        <input
          aria-invalid="false"
          autocomplete="new-password"
          class="amplify-field-group__control amplify-field__show-password amplify-input"
          id="mockId"
          name="password"
          placeholder=""
          required=""
          type="password"
        />
      </div>
      <div
        class="amplify-field-group__outer-end"
      >
        <button
          amplify-button=""
          aria-checked="false"
          aria-label="Show password"
          class="amplify-field-group__control amplify-input"
          role="switch"
        >
          <svg
            class="amplify-icon"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
          
          
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`amplify-password-field renders as expected when show password button is clicked 1`] = `
<div
  id="root1"
>
  <div
    class="amplify-field amplify-flex amplify-passwordfield amplify-textfield"
  >
    <label
      class="amplify-label"
      for="mockId"
    >
       Password 
    </label>
    <div
      class="amplify-field-group amplify-flex"
    >
      <div
        class="amplify-field-group__field-wrapper"
      >
        <input
          aria-invalid="false"
          autocomplete="new-password"
          class="amplify-field-group__control amplify-field__show-password amplify-input"
          id="mockId"
          name="password"
          placeholder=""
          required=""
          type="text"
        />
      </div>
      <div
        class="amplify-field-group__outer-end"
      >
        <button
          amplify-button=""
          aria-checked="true"
          aria-label="Hide password"
          class="amplify-field-group__control amplify-input"
          role="switch"
        >
          
          <svg
            class="amplify-icon"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
          
        </button>
      </div>
    </div>
  </div>
</div>
`;
